### PR TITLE
print tensor information (enhancement)

### PR DESCRIPTION
**Pull request by mohamed-180**
`> x <- torch_randn(5)`
> x
torch_tensor
 0.0341
 1.2221
 1.2651
-0.3475
 0.2250
[ CPUFloatType{5} ]

`> x$requires_grad_()`
>torch_tensor
 0.0341
 1.2221
 1.2651
-0.3475
 0.2250
[ CPUFloatType{5} ][ requires_grad = TRUE ]

`> y <- torch_log(x)`
`> y`
>torch_tensor
-3.3771
 0.2005
 0.2352
    nan
-1.4916
[ CPUFloatType{5} ][ grad_fn = LogBackward ]

### CODE REVIEW

File: R/tensor.R
Comment (dfalbel) [inline R/tensor.R:51]: Maybe we need a `cat("\n")` after everything to have the same effect as `std::endl`? So the next thing printed to the console starts in its own line?

```suggestion
      }
      cat("\n")
```

**Review by dfalbel — CHANGES_REQUESTED** [review]
Thanks @mohamed-180 ! Looks great! I have submitted a new torch version to CRAN, so I'll just wait for it to be accepted before we merge this PR. 

In the meantime, maybe you want to add a test case. You could use something like:

```
test_that("print tensors with grad_fn and requires_grad", {
  testthat::local_edition(3)
  expect_snapshot(print(x))
})
```

File: R/tensor.R
Comment (mohamed-180) [reply R/tensor.R:51]: done

**Comment by dfalbel** [comment]
Thanks very much @mohamed-180 ! I moved your PR's to #673 and will merge as soon as tests passes.

**Comment by dfalbel** [comment]
It's now merged! Thanks very much @mohamed-180 !

**Pull request merged**
